src/kolibri2zim/scraper.py

```diff
@@ -197,21 +193,24 @@ def schedule_node(item):
 
     def add_node(self, item):
         """process a content node from the tuple in queue"""
-        node_id, kind = item
-        # check if we have a handler for this {kind} of node
-        handler = getattr(self, f"add_{kind}_node", None)
+        try:
```

Comment:
I don't get why you're putting all this safe code into the try block. try block of an entire function is not a good pattern

Reply:
The goal is to wrap any exception which might occur so that we are sure to have both nodeid and kind in the logs.

I agree that most of this code is probably safe, errors will probably only happen in the `handler` or `funnel_file` methods, but I assume that "we never know".

Should I move the try block to a `safe_add_node` method which will call the `add_node`? (kind of hiding the thing, but probably easier to read) Or is it better to move the try block only around the code inside the `if handler ...`?

Reply:
What about a decorator?

Reply:
Of course